dll/shellext/zipfldr/CZipCreator.cpp

```diff
@@ -65,7 +58,10 @@ DoGetNameInZip(const CStringW& basename, const CStringW& filename)
 
     ret.Replace(L'\\', L'/');
 
-    return DoGetAnsiName(ret);
+    if (!bUtf8)
+        return CStringA(ret);
+
+    return CStringA(CW2AEX<MAX_PATH>(ret, CP_UTF8));
```

Comment:
Just to recall, if the necessary internal buffer for the conversion happens to be larger than the default `MAX_PATH`, it would be reallocated with corret size?

Reply:
When the source string was too long, then It will use a heap memory.